From 572399babc22fde61f25b17d264ead8b1374fa94 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 05 Jul 2021 13:06:28 +0300
Subject: [PATCH] [SW-46517] preboot: greco: remove led on support

Currently due to GPIO2 is unset as part of led ON function,
it causes problem in running secureboot flow after reset as
arcboot ROM checks this GPIO2 and execute recovery flow if
its unset. Till this issue is fixed, disable LED on support
from preboot.

Change-Id: I59c52f916ee7299b094e95063fb7eccae037b1df
---

diff --git a/arch/greco_arc_gnu/asic.c b/arch/greco_arc_gnu/asic.c
index 835845a..19a576a 100644
--- a/arch/greco_arc_gnu/asic.c
+++ b/arch/greco_arc_gnu/asic.c
@@ -44,15 +44,7 @@
 /* configure yellow LED(GPIO2) in output mode and turn on(active low) */
 void preboot_led_on(void)
 {
-	u32 val = BIT(2);
-
-	/* configure GPIO in output mode */
-	regmap_writel(val, mmPSOC_GPIO0_OUTENSET);
-
-	/* Turn ON LED (pin is active low) */
-	val = regmap_readl(mmPSOC_GPIO0_DATAOUT);
-	val &= ~BIT(2);
-	regmap_writel(val, mmPSOC_GPIO0_DATAOUT);
+	/* TODO: perform LED GPIO configuration */
 }
 
 u32 hl_spi_baudr_get(void)
